src/drt/src/TritonRoute.cpp

```diff
@@ -515,3 +932,276 @@ void TritonRoute::setParams(const ParamStruct& params)
     MINNUMACCESSPOINT_MACROCELLPIN = params.minAccessPoints;
   }
 }
+
+void TritonRoute::addWorkerResults(
+    const std::vector<std::pair<int, std::string>>& results)
+{
+  std::unique_lock<std::mutex> lock(results_mutex_);
+  workers_results_.insert(
+      workers_results_.end(), results.begin(), results.end());
+  results_sz_ = workers_results_.size();
+}
+
+bool TritonRoute::getWorkerResults(
+    std::vector<std::pair<int, std::string>>& results)
+{
+  std::unique_lock<std::mutex> lock(results_mutex_);
+  if (workers_results_.empty())
+    return false;
+  results = workers_results_;
+  workers_results_.clear();
+  results_sz_ = 0;
+  return true;
+}
+
+int TritonRoute::getWorkerResultsSize()
+{
+  return results_sz_;
+}
+
+void TritonRoute::reportDRC(const string& file_name, FlexDRWorker* worker)
+{
+  double dbu = getDesign()->getTech()->getDBUPerUU();
+
+  if (file_name == string("")) {
+    if (VERBOSE > 0) {
+      logger_->warn(
+          DRT,
+          290,
+          "Waring: no DRC report specified, skipped writing DRC report");
+    }
+    return;
+  }
+
+  ofstream drcRpt(file_name.c_str());
+  if (drcRpt.is_open()) {
+    for (auto& marker : getDesign()->getTopBlock()->getMarkers()) {
+      // get violation bbox
+      Rect bbox;
+      marker->getBBox(bbox);
+      if (worker != nullptr && !worker->getDrcBox().intersects(bbox))
+        continue;
+      auto con = marker->getConstraint();
+      drcRpt << "  violation type: ";
+      if (con) {
+        switch (con->typeId()) {
+          case frConstraintTypeEnum::frcShortConstraint: {
+            if (getDesign()
+                    ->getTech()
+                    ->getLayer(marker->getLayerNum())
+                    ->getType()
```

Comment:
Just call ...->getType() once and then if-else on the result.

Reply:
I don't think this is related to this PR specifically because I just copied this function from FlexDR and exposed it to TritonRoute, but sure will do.